jax/_src/lax/lax.py

```diff
@@ -83,9 +83,15 @@
 
 T = TypeVar("T")
 
+def _validate_shapes(shapes: Sequence[Tuple[int, ...]]):
+  assert shapes
+  for shape in shapes:
+    if not all(np.issubdtype(type(ix), np.integer) and ix >= 0 for ix in shape):
```

Comment:
This subdtype check is not correct – for example, dimensions can be instances of `jax.numpy.int32`, or they can be dynamic size objects. Better would be to call `_canonicalize_dimension` on each entry, because it takes care of these details (and raises an appropriate error if the dimension is invalid)

Reply:
Ah, I got you. Supposedly, this also handles the case that shape is a traced value? I'll try and fix it later in the day.